lib/src/model/settings/general_preferences.dart

```diff
@@ -82,13 +108,16 @@ class GeneralPrefsState with _$GeneralPrefsState {
 
     /// Should enable system color palette (android 12+ only)
     required bool systemColors,
+    @JsonKey(toJson: localeToJson, fromJson: localeFromJson)
+    required Locale? locale,
   }) = _GeneralPrefsState;
 
   static const defaults = GeneralPrefsState(
     themeMode: ThemeMode.system,
     isSoundEnabled: true,
     soundTheme: SoundTheme.standard,
     systemColors: true,
+    locale: null, // use system locale
```

Comment:
Instead of this inline comment, prefer a doc comment above the field declaration.